pylds/base.py

```diff
@@ -186,18 +285,32 @@ def compute_lagrangian_descriptor(grid_parameters, vector_field, tau, p_value=0.
     LD : ndarray, shape (Nx, Ny)
         Array of computed Lagrangian descriptor values for all initial conditions.
     """
+    #get visualisation slice parameters and Number of DoF
+    if type(grid_parameters) == dict:
+        #n-DoF systems
+        slice_parameters = grid_parameters['slice_parameters'] # 2n-D grid
+        N_dim = len(grid_parameters['dof_slice'])
+    else:
+        #1-DoF systems
+        slice_parameters = grid_parameters # 2-D grid
+        N_dim = len(slice_parameters)
+        
+    #set boundaries for escape-box condition, if not defined
+    if not box_boundaries:
+        box_boundaries = int(N_dim/2)*[[-np.infty, np.infty]] #restricted to configuration space
     
-    f = lambda x, y: vector_field_flat(x, y, vector_field, p_value, box_boundaries)
-    y0 = generate_points(grid_parameters)
+    #solve initial value problem
+    f = lambda t, y: vector_field_flat(t, y, vector_field, p_value, box_boundaries)
+    y0, mask = generate_points(grid_parameters)
     
     solution = solve_ivp(f, [0,tau], y0, t_eval=[tau], rtol=1.0e-4)
```

Comment:
Is the mask never passed to the integrator? In that case there is potentially a lot of computing time wasted on initial conditions that do not satisfy the energy condition and are ultimately thrown away.

Reply:
I need to check if I can pass an object defined via `np.ma`, so that the code would look like this
```
y0, mask = generate_points(grid_parameters)
y0 = np.ma(y0, mask=mask)
solution = solve_ivp(f, [0,tau], y0, t_eval=[tau], rtol=1.0e-4)
```

I just assumed that `y0` must be always an array or a matrix. I'll have to check it out.

Reply:
Sorted. Applied mask to `y0` values too.

HOWEVER, no noticeable effect on speed was noticeable when I run the code with/without masked `y0` values. I ran each version multiple times (up to 5) and execution times were kept around 35s, Henon-Heiles system. Perhaps for this case there no noticeable effect. I implemented the solution anyway just in case.

Reply:
Excellent. The effect should become noticable once you start messing around with location of the slice and energy, for example a slice of all nans. It will also make debugging for users a lot simpler - conservation of energy in the integrator is probably the first property to look at. This way only trajectories at the predefined energy will be present in the calculations.